flutter/shell/platform/tizen/channels/platform_channel.cc

```diff
@@ -140,6 +153,33 @@ void PlatformChannel::HandleMethodCall(
     document.AddMember(rapidjson::Value(kValueKey, allocator),
                        rapidjson::Value(ClipboardHasStrings()), allocator);
     result->Success(document);
+  } else if (method == kExitApplicationMethod) {
+    rapidjson::Value::ConstMemberIterator iter =
+        arguments->FindMember(kExitTypeKey);
+    if (iter == arguments->MemberEnd()) {
+      result->Error(kExitRequestError, "Invalid application exit request.");
+      return;
+    }
+
+    const std::string& exit_type = iter->value.GetString();
+    rapidjson::Document document;
+    document.SetObject();
+    if (!initialization_complete_ || exit_type == kExitTypeRequired) {
+      ui_app_exit();
+      document.AddMember(kExitResponseKey, kExitResponseExit,
+                         document.GetAllocator());
+      result->Success(document);
```

Comment:
Question: Is it okay to execute any code after the `ui_app_exit` call?

Reply:
As far as I know, Embedder's Shutdown() is called when ui_app_exit() is called and terminate() of service_app_lifecycle_callback_s is called.

But I don't know if `result->Success()` is always guaranteed to be called because we don't know when the terminate() cb will be called.
However, when I checked with the log, the code after ui_app_exit() was called.

Reply:
I'm not sure if the terminate callback is called directly by the `ui_app_exit` function. If it is called synchronously, the engine will shut down immediately and the engine's message channel and the instance of this class (`PlatformChannel`) will become invalid right after the `ui_app_exit` call. If you didn't see an error like "Invalid engine handle" or the app didn't crash (SIGABRT), this should not be the case (i.e. the terminate callback is not called synchronously). We still don't know what's done internally by `ui_app_exit`.

Reply:
@swift-kim When we call ui_app_exit(), ui_app_lifecycle_callback_s.terminate() is called at the end of the main loop.
[The documentation](https://docs.tizen.org/iot/api/5.0/tizen-iot-headed/group__CAPI__APPLICATION__MODULE.html#ga3f00a008e3612d97f68042635203d792) recommends deleting resources here.

terminate() calls FlutterDesktopViewDestroy() and the associated destructors are called sequentially.
The method call to Success() is expected to be processed before that. There were no related error messages or logs.

Please let me know if there is anything else I need to consider regarding this patch.